compiler/rustc_target/src/asm/xtensa.rs

```diff
@@ -0,0 +1,276 @@
+        Ok(())
+    }
+}
+
+def_regs! {
+    Xtensa XtensaInlineAsmReg XtensaInlineAsmRegClass {
+        a2: reg = ["a2"],
+        a3: reg = ["a3"],
+        a4: reg = ["a4"],
+        a5: reg = ["a5"],
+        a6: reg = ["a6"],
+        a7: reg = ["a7"] % frame_pointer_a7,
+        a8: reg = ["a8"],
+        a9: reg = ["a9"],
+        a10: reg = ["a10"],
+        a11: reg = ["a11"],
+        a12: reg = ["a12"],
+        a13: reg = ["a13"],
+        a14: reg = ["a14"],
+        a15: reg = ["a15"] % frame_pointer_a15,
+        sar: reg = ["sar"],
+        configid0: reg = ["configid0"], 
+        configid1: reg = ["configid1"],
+        lbeg: reg = ["lbeg"] % has_loop,
+        lend: reg = ["lend"] % has_loop,
+        lcount: reg = ["lcount"] % has_loop,
+        litbase: reg = ["litbase"] % has_extendedl32r,
+        scompare1: reg = ["scompare1"] % has_s32c1i,
+        acclo: reg = ["acclo"] % has_mac16,
+        acchi: reg = ["acchi"] % has_mac16,
+        m0: reg = ["m0"] % has_mac16,
+        m1: reg = ["m1"] % has_mac16,
+        m2: reg = ["m2"] % has_mac16,
+        m3: reg = ["m3"] % has_mac16,
+        windowbase: reg = ["windowbase"] % has_windowed,
+        windowstart: reg = ["windowstart"] % has_windowed,
+        ddr: reg = ["ddr"] % has_debug,
+        ibreakenable: reg = ["ibreakenable"] % has_debug,
+        ibreaka0: reg = ["ibreaka0"] % has_debug,
+        ibreaka1: reg = ["ibreaka1"] % has_debug,
+        dbreaka0: reg = ["dbreaka0"] % has_debug,
+        dbreaka1: reg = ["dbreaka1"] % has_debug,
+        dbreakc0: reg = ["dbreakc0"] % has_debug,
+        dbreakc1: reg = ["dbreakc1"] % has_debug,
+        icount: reg = ["icount"] % has_debug,
+        icountlevel: reg = ["icountlevel"] % has_debug,
+        debugcause: reg = ["debugcause"] % has_debug,
+        memctl: reg = ["memctl"] % has_memctl,
+        atomctl: reg = ["atomctl"] % has_atomctl,
+        ps: reg = ["ps"] % has_exception,
+        epc1: reg = ["epc1"] % has_exception,
+        epc2: reg = ["epc2"] % has_exception,
+        epc3: reg = ["epc3"] % has_exception,
+        epc4: reg = ["epc4"] % has_exception,
+        epc5: reg = ["epc5"] % has_exception,
+        epc6: reg = ["epc6"] % has_exception,
+        epc7: reg = ["epc7"] % has_exception,
+        depc: reg = ["depc"] % has_exception,
+        eps2: reg = ["eps2"] % has_exception,
+        eps3: reg = ["eps3"] % has_exception,
+        eps4: reg = ["eps4"] % has_exception,
+        eps5: reg = ["eps5"] % has_exception,
+        eps6: reg = ["eps6"] % has_exception,
+        eps7: reg = ["eps7"] % has_exception,
+        excsave1: reg = ["excsave1"] % has_exception,
+        excsave2: reg = ["excsave2"] % has_exception,
+        excsave3: reg = ["excsave3"] % has_exception,
+        excsave4: reg = ["excsave4"] % has_exception,
+        excsave5: reg = ["excsave5"] % has_exception,
+        excsave6: reg = ["excsave6"] % has_exception,
+        excsave7: reg = ["excsave7"] % has_exception,
```

Comment:
I think this might be a bug in the llvm backend, reported [here](https://github.com/espressif/llvm-project/issues/47#issuecomment-895823989), if it is then I think I can just use the `has_highpriinterrupts` features to mark those registers unavailable on the esp8266?

Reply:
Yes, it might be something we should fix in LLVM first, then. A boolean `HasHighPriInterrupts` flag also doesn't exactly define the number of interrupt levels as one could theoretically build an Xtensa chip with 2 low-medium interrupts and 1 high interrupt... Let's ignore this for now since we don't actually have such hardware.